helper/kvflag.go

```diff
@@ -22,11 +22,38 @@ func (v *Flag) Set(raw string) error {
 	}
 
 	if *v == nil {
-		*v = make(map[string]string)
+		*v = make(map[string]interface{})
 	}
 
-	key, value := raw[0:idx], raw[idx+1:]
-	(*v)[key] = value
+	// Split the variable key based on the nested delimiter to get a list of
+	// nested keys.
+	keys := strings.Split(raw[0:idx], ".")
+
+	// If we only have a single key, then we are not dealing with a nested set
+	// meaning we can update the variable mapping and exit.
+	if len(keys) == 1 {
+		(*v)[keys[0]] = raw[idx+1:]
+		return nil
+	}
+
+	// Identify the index max of the list for easy use.
+	nestedLen := len(keys) - 1
+
+	// The end map is the only thing we concretely know which contains our
+	// final key:value pair.
+	endEntry := map[string]interface{}{keys[nestedLen]: raw[idx+1:]}
+
+	// Track the root of the nested map structure so we can continue to iterate
+	// the nested keys below.
+	root := endEntry
+
+	// Iterate the nested keys backwards. Set a new root map containing the
+	// previous root as its value. Do not iterate backwards fully to the end,
+	// instead save the first key for the entry into Flag.
+	for i := nestedLen - 1; i > 0; i-- {
+		root = map[string]interface{}{keys[i]: root}
```

Comment:
i think that this (and https://github.com/hashicorp/levant/pull/382/files#diff-ab81c85d7e7eb24d04413d924ece8d962e1e662b38e20de322187771054a5d10R44 above) need to consider any existing content. 

see below:
```
$ cat test.tmpl
[[.siblings.a]]
[[.siblings.b]]

$ ./bin/levant render -var "siblings.a=bob" -var "siblings.b=alice"  test.tmpl
<no value>
alice

$ ./bin/levant render -var "siblings.b=alice"  -var "siblings.a=bob" test.tmpl
bob
<no value>
```

perhaps this loop needs run from `0` to `nestedLen`, diving into `*v` and creating a new map only when a key doesn't exist.

Reply:
i can update this PR if you like.

Reply:
If you have scope that would be great as I won't be getting around to this until the morning now.

Reply:
@jrasell , added a new commit. i wasn't sure what the behavior should be if there were conflicting `-var`; currently, `-var a=1 -var a=2` is last-one-wins. maybe this should be the same, but i went ahead and added an error case if the user provides `-var a=1 -var a.b=2`, from the argument that we should let them know if they're doing something silly. will defer to your opinion on this.

Reply:
That makes sense to me. I can't think of a situation where overwrite behaviour in this situation would really be desired rather than being an error.

As I raised the PR I cannot approve it or the changes, so will need you to do this, pretty please if you're happy.